tests/checks/basic.fish

```diff
@@ -625,8 +652,24 @@ $fish -c 'echo \ufdd2"fart"'
 echo (printf '\ufdd2foo') | string escape
 # CHECK: \Xef\Xb7\X92foo
 
-printf '%s\n' "#!/bin/sh" 'echo $0' > $tmpdir/argv0.sh
+printf '%s\n' "#!/bin/sh" 'echo $0' >$tmpdir/argv0.sh
 chmod +x $tmpdir/argv0.sh
 cd $tmpdir
 ./argv0.sh
 # CHECK: ./argv0.sh
+
+# A simple check of badly formatted code
+#!fish_indent: off
+  begin; echo -n "1"
+# A comment
+   ; echo -n          2 |
+cat (
+		
+	echo -n 3|psub
+)
+
+        set a 4 &&
+echo -n $a[      1 ]
+    echo -n 5         >         /dev/stdout; end
+#!fish_indent: on
+#CHECK: 1345
```

Comment:
Nice test - I'd expect it to live in tests/checks/indent.fish,
we should be able to simply copy it over